pandas/tseries/period.py

```diff
@@ -940,7 +916,8 @@ def _wrap_union_result(self, other, result):
 
     def _apply_meta(self, rawarr):
         if not isinstance(rawarr, PeriodIndex):
-            rawarr = PeriodIndex(rawarr, freq=self.freq)
+            rawarr = PeriodIndex._from_ordinals(rawarr, freq=self.freq,
```

Comment:
Yes I only saw it because it broke a test. But not really sure whether we need the function